pkg/aws/core_util.go

```diff
@@ -79,27 +79,18 @@ func disableSrcAndDestCheck(svc ec2iface.EC2API, instanceID *string) error {
 	return nil
 }
 
-// getInstancesFromMachineName extracts AWS Instance object from given machine name
-func (d *Driver) getInstancesFromMachineName(machineName string, providerSpec *api.AWSProviderSpec, secret *corev1.Secret) ([]*ec2.Instance, error) {
+func getMachineInstancesByTagsAndStatus(svc ec2iface.EC2API, machineName string, providerSpecTags map[string]string) ([]*ec2.Instance, error) {
```

Comment:
I know that this is an existing piece of code but nevertheless i have a question.
If the tags for cluster and node role are missing then what will describeInstance return?
Will will the impact?

Reply:
If the tags for cluster or nodeRole are missing, then `getMachineInstancesByTagsAndStatus` does not return an error, but returns a nil slice of instances. So we will try to fetch the instance by its ID.

Reply:
We set the cluster and node role to empty strings. This results in an empty slice, and the fallback is tried.

Reply:
ok then we are good. Thanks for testing this.